src/main/java/com/github/kokorin/jaffree/ffmpeg/FFmpeg.java

```diff
@@ -363,20 +365,49 @@ public FFmpegResult execute() {
      * @return ffmpeg result future
      */
     public FFmpegResultFuture executeAsync() {
+        return executeAsync(new Executor() {
+            @Override
+            public void execute(Runnable command) {
+                Thread runner = new Thread(command, "FFmpeg-async-runner");
+                runner.setDaemon(true);
+                runner.start();
+            }
+        });
+    }
+    
+    /**
+     * Starts asynchronous ffmpeg execution, executed using the supplied Executor
+     *
+     * @param executor the executor to use for asynchronous execution
+     * @return ffmpeg result future
+     */
+    public FFmpegResultFuture executeAsync(java.util.concurrent.Executor executor) {
         final ProcessHandler<FFmpegResult> processHandler = createProcessHandler();
         Stopper stopper = createStopper();
         processHandler.setStopper(stopper);
 
-        FutureTask<FFmpegResult> resultFuture = new FutureTask<>(new Callable<FFmpegResult>() {
+        CompletableFuture<FFmpegResult> resultFuture = new CompletableFuture<FFmpegResult>() {
             @Override
-            public FFmpegResult call() throws Exception {
-                return processHandler.execute();
+            public boolean cancel(boolean mayInterruptIfRunning) {
+                if (mayInterruptIfRunning) {
+                    stopper.forceStop();
+                } else {
+                    stopper.graceStop();
+                }
+                return completeExceptionally(new CancellationException());
             }
-        });
+        };
 
-        Thread runner = new Thread(resultFuture, "FFmpeg-async-runner");
-        runner.setDaemon(true);
-        runner.start();
+        Objects.requireNonNull(executor, "Executor service must be provided").execute(new Runnable() {
```

Comment:
Parameters should be verified in the beginning. Personally I would omit this check.
The string is too long, please run checkstyle.